lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -528,18 +530,31 @@ type installKubeAgentParams struct {
 	req          EnrollEKSClustersRequest
 }
 
-func getChartURL(version string) *url.URL {
+func getChartURL(version string) (*url.URL, error) {
+	repo := agentRepoURL
+	ver, err := semver.NewVersion(version)
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to parse chart version %q", version)
+	}
+
+	// pre release tagged charts are located in the staging repo.
+	if ver.PreRelease != "" {
```

Comment:
FYI `-cloud.X` releases still go to production env. We may need to account for it here because we deploy those to Cloud sometimes.

Reply:
Actually, I take it back. I don't think we publish Helm charts for -cloud releases at all currently, only Docker images.

Reply:
Should we I future-proof it and still add an exception for "cloud" tag?

Reply:
in that case, it will fail because the chart is missing and we should drop the cloud.x

Reply:
Yeah I'm thinking what is the best way to handle -cloud releases here. My first idea was that we can fallback to previous full version in this case so if control plane is running say 15.3.4-cloud.X, we use 15.3.3. But this might get messy with .0 releases, even though it's an edge case.

Reply:
can't we just use whatever is available in the stable release channel?

Reply:
As discussed in Slack, this might not be an issue since in Cloud we take versions from the bucket lock file when auto-upgrades are enabled.